src/mbed_tools/build/config.py

```diff
@@ -12,6 +12,10 @@
 from mbed_tools.build._internal.write_files import write_file
 
 
+DEFAULT_CONFIG_DIR = ".mbedbuild"
```

Comment:
In the commit description for "config: Write config file to target subdirectory "...

> Writes generated config files to the specific target/develop/toolchain/
subdirectory as well as .mbedbuild.

Could you please explain why it is good to put the generated config files into both places? It might be good to remark explicitly on why we aren't putting the generated config files to only the specific subdirectory.